lib/v2/screens/profile_screens/citizenship/interactor/mappers/set_values_mapper.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:seeds/v2/datasource/remote/model/profile_model.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/v2/screens/profile_screens/citizenship/interactor/viewmodels/citizenship_state.dart';
+
+class SetValuesStateMapper extends StateMapper {
+  CitizenshipState mapResultToState(CitizenshipState currentState, Result result) {
+    if (result.isError) {
+      return currentState.copyWith(pageState: PageState.failure, errorMessage: result.asError!.error.toString());
+    } else {
+      double timeline = 0;
+      List<ProfileModel> profiles = result.asValue?.value as List<ProfileModel>;
+
+      // Define timeline
+      if (currentState.profile?.status == ProfileStatus.visitor) {
+        // Timeline to resident
+        // If the scores are greater than those required, use the required ones instead to avoid errors in the formula.
+        int reputation = currentState.score!.reputationScore! > resident_required_reputation
+            ? resident_required_reputation
+            : currentState.score!.reputationScore!;
+        int visitors = profiles.isNotEmpty ? resident_required_visitors_invited : 0;
+        int planted = currentState.score!.plantedScore! > resident_required_planted_seeds
+            ? resident_required_planted_seeds
+            : currentState.score!.plantedScore!;
+        int transactions = currentState.score!.transactionsScore! > resident_required_seeds_transactions
+            ? resident_required_seeds_transactions
+            : currentState.score!.transactionsScore!;
+
+        // Timeline to resident formula
+        timeline = ((reputation / resident_required_reputation) +
+                (visitors / resident_required_visitors_invited) +
+                (planted / resident_required_planted_seeds) +
+                (transactions / resident_required_seeds_transactions)) /
+            4 *
+            100;
+      } else {
+        // Timeline to citizen
+        // If the scores are greater than those required, use the required ones instead to avoid errors in the formula.
+        int reputation = currentState.score!.reputationScore! > citizen_required_reputation
```

Comment:
there is a lot if (!) use in here. 
1 - Can we think of a better way to model our states to avoid this? 

2 - Should we do some checks before using the values? 

The user of (!) should be last resource.

Reply:
I dhange the score model so that its fields are required now, and create a profile and score not null variables before operating on the mapper, in order to avoid the repetitive use of (!).